sql/core/src/main/java/org/apache/spark/sql/execution/vectorized/WritableColumnVector.java

```diff
@@ -846,7 +849,14 @@ public final void addElementsAppended(int num) {
   /**
    * Marks this column as being constant.
    */
-  public final void setIsConstant() { isConstant = true; }
+  public final void setIsConstant() {
+    if (childColumns != null) {
+      for (WritableColumnVector c : childColumns) {
+        c.setIsConstant();
```

Comment:
This change make sense. Do you know why there was no problem before?

Reply:
For example, `alter table t add column s array<int> default array(1, 2)` , spark will create a vector for column s and , and a vector for the items of this column.
Before this PR, both those two vectors will not be reset.
After this PR, the second vector will be reset without this change. 

![image](https://github.com/apache/spark/assets/3626747/e77926f3-6ac0-4cf5-a776-6e7b3f0957ef)